app/src/lib/git/cherry-pick.ts

```diff
@@ -302,13 +305,28 @@ export async function getCherryPickSnapshot(
 export async function continueCherryPick(
   repository: Repository,
   files: ReadonlyArray<WorkingDirectoryFileChange>,
+  manualResolutions: ReadonlyMap<string, ManualConflictResolution> = new Map(),
   progressCallback?: (progress: ICherryPickProgress) => void
 ): Promise<CherryPickResult> {
   // only stage files related to cherry pick
   const trackedFiles = files.filter(f => {
     return f.status.kind !== AppFileStatusKind.Untracked
   })
-  await stageFiles(repository, trackedFiles)
+
+  // apply conflict resolutions
+  for (const [path, resolution] of manualResolutions) {
+    const file = files.find(f => f.path === path)
+    if (file !== undefined) {
+      await stageManualConflictResolution(repository, file, resolution)
+    } else {
+      log.error(
+        `[continueCherryPick] couldn't find file ${path} even though there's a manual resolution for it`
+      )
+    }
```

Comment:
As a general rule I prefer to avoid `else` and extra nesting when handling these kinds of checks:
```suggestion
    if (file === undefined) {
      log.error(
        `[continueCherryPick] couldn't find file ${path} even though there's a manual resolution for it`
      )
      continue
    }

    await stageManualConflictResolution(repository, file, resolution)
```

Basically, I usually try to turn error checks into "fast returns" (or fast `continue`s in this case 🤣 ), leaving the "happy path" in a clearer state.

What do you think?

Reply:
Yep. I do too.. :) copy/pasta from rebase and merge. Random side note, on my last team, I got the opposite PR comments to undo fast returns. This comment makes me happy! 😄